### DISCUSSION TITLE
Only Office integration Failed

### DISCUSSION THREAD

**Discussion by sergeByishimo** [exoplatform/exo-community#37] [Q&A]
I have issues related to the [onlyoffice-addon](https://github.com/exoplatform/onlyoffice)

PS: README file  is 4 years old

1. Is this compatibility matrix still valid for the addon newer versions like `2.4.2`, `2.4.3`? whats the current compatibility matrix with onlyoffice documents server?
<img width="730" alt="Screen Shot 2023-09-20 at 10 52 47" src="https://github.com/exoplatform/exo-community/assets/32865999/2d9bd690-69b9-46a0-9663-4e9c9fe9ed19">

Onlyoffice version `5.4.2.46` is from 2019 I believe, I integrated it in exo platform `6.4.2` and I'm getting this error when trying to open a document : `The "documentType" parameter for the config object is invalid. Please correct it.`

2. Now also using only-office document server latest version `7.4.1` : I'm getting this error, when trying to open a file from exo space documents  : `Download Failed` . Keep in mind that I put in the configuration mentioned in the README file  : `onlyoffice.documentserver.accessOnly=false`  to allow requests from any client host (e.g. for development or in case of proxy/NAT settings), but still get the error

**Comment by rdenarie** [comment]
Hello @sergeByishimo 
For your first question, I just create a pr to update compatibility matrix
Result is :
![image](https://github.com/exoplatform/exo-community/assets/807839/3ec97baf-54de-4523-a04b-d55c4aadf02a)

With eXo 6.4.2 you should use onlyffice 7.2.2

**Reply by sergeByishimo** [reply]
I did that!  I used  onlyffice `7.2.2` in eXo `6.4.2` and It generated this very short secret key, which I used in exo configuration `onlyoffice.documentserver.secret` and it gave me this error in exo logs :
`
Caused by: io.jsonwebtoken.security.WeakKeyException: The specified key byte array is 96 bits which is not secure enough for any JWT HMAC-SHA algorithm.  The JWT JWA Specification (RFC 7518, Section 3.2) states that keys used with HMAC-SHA algorithms MUST have a size >= 256 bits (the key size must be greater than or equal to the hash output size).  Consider using the io.jsonwebtoken.security.Keys#secretKeyFor(SignatureAlgorithm) method to create a key guaranteed to be secure enough for your preferred HMAC-SHA algorithm.  See https://tools.ietf.org/html/rfc7518#section-3.2 for more information.
`

 the addon couldnt even preview a file using onlyoffice version `7.2.2`.


<img width="1477" alt="Screen Shot 2023-09-20 at 19 18 52" src="https://github.com/exoplatform/exo-community/assets/32865999/c518e8d4-3875-401f-aca2-ea40c7249b69">

<br><br>

And when I used onlyoffice `7.4.1` , it generated a longer secret key and there was no key error, but instead I get the "The document could not be saved error" when I open the editor to edit a file, but it can preview a file.

<img width="1505" alt="Screen Shot 2023-09-20 at 19 42 36" src="https://github.com/exoplatform/exo-community/assets/32865999/ca4d9998-ad53-482a-acca-0bc4a7c56ebb">

<img width="1477" alt="Screen Shot 2023-09-20 at 18 52 59" src="https://github.com/exoplatform/exo-community/assets/32865999/becbe975-09bf-4b16-b8a5-af75056a8dcd">

**Comment by rdenarie** [comment]
Hello. 
Sorry for the late delay, I have to check informations before answering you.

Firstly, if you use exo 6.4, keep OO 7.2.2.

You can change the OO secret in OO container by editing this file : 
/etc/onlyoffice/documentserver/local.json
Mount it from your host to your container, so that you can modify it.

Then, in eXo, you can add this property in {exoHome}/gatein/conf/exo.properties : 
onlyoffice.documentserver.secret=xxxxxxx

Replace "xxxxxxx" by the secret you put in the json oo file.

The secret must have size >= 256 bits.
